src-plugins/mscAlgorithmPaint/mscAlgorithmPaintToolBox.cpp

```diff
@@ -604,18 +605,29 @@ void AlgorithmPaintToolBox::import()
 
 void AlgorithmPaintToolBox::updateView()
 {
+    setButtonsDisabled(true);
     medAbstractView *view = this->getWorkspace()->stackedViewContainers()->getFirstSelectedContainerView();
 
     if (view)
     {
-        setCurrentView(qobject_cast<medAbstractImageView*>(view));
-
-        updateMouseInteraction();
-
-        medAbstractData* data = currentView->layerData(0);
-
-        if(data)
+        medAbstractImageView * v = qobject_cast<medAbstractImageView*>(view);
+        if (v)
         {
+            // AlgorithmPaintToolBox does not work with meshes
+            for (unsigned int i=0 ; i<v->layersCount() ; ++i)
+            {
+                medAbstractData *data = v->layerData(i);
+                if(!data || data->identifier().contains("vtkDataMesh"))
+                {
+                    qDebug()<<"AlgorithmPaintToolBox::updateView() do not use meshes in this toolbox.";
+                    return;
+                }
```

Comment:
Since this pb can appear in other toolboxes (polygonRoiToolbox for instance), maybe we can try to set this toolbox behaviour in ```medToolBox``` as we did [here](https://github.com/Inria-Asclepios/medInria-public/commit/5b3332338bcc933adfa8c73e750e6196c80d0222) for the "inputs must be the same size" problem.

It means creating a new ```medAbstractProcess::DataError``` (```NO_MESH``` for instance), modifying ```medToolBox::handleDisplayError``` to manage this error.

Then in AlgorithmPaintToolBox, you can directly use this latter method : ```handleDisplayError(medAbstractProcess::NO_MESH); ```

Eventually, you could also modify polygonROIToolBox so that its ```update()```method returns ```NO_MASK``` when facing this problem.

It might seem tricky but it's quite easy to understand with the link I gave above.